### ISSUE TITLE
DHTEndpoint improvements

### ISSUE THREAD

**Issue by qstokkink** [Tribler/py-ipv8#451]
Migrated from https://github.com/Tribler/tribler/issues/4227#issuecomment-464483808 by @egbertbouman, regarding the REST `DHTEndpoint` :

>Looking at this code I do wonder about the chances of everything going OK. I see no retry mechanism, and the blocklistener seems to get notified only once about a new block.. Also note that values in the DHT are currently not automatically republished, meaning that the value will dissapear after relatively short period.

> Additionally, if the number of chunks grow beyond MAX_VALUES_IN_FIND, you'll never be able to get the full block from the DHT. Then again, blocks are probably small.

**Comment by DanGraur** [comment]
I think I can take a look at this.

**Comment by qstokkink** [comment]
I believe this issue has been sufficiently addressed now (and the OP is too confusing to keep this one open). @DanGraur if there is anything left to do for the DHTEndpoint please make a new issue.

**Comment by DanGraur** [comment]
Sure thing!
